[BUILD] Build break on macOS 12

The github macOS-latest label has changed from macOS-11 to macOS-12

https://github.com/actions/runner-images/issues/6384

This exposes a new build break when compiling opentelemetry-cpp with all dependencies:

```
external/com_google_absl/absl/functional/function_ref.h:124:16: error: definition of implicit copy constructor for 'FunctionRef<void (absl::string_view)>' is deprecated because it has a user-declared copy assignment operator [-Werror,-Wdeprecated-copy]
  FunctionRef& operator=(const FunctionRef& rhs) = delete;
               ^
external/com_google_absl/absl/strings/cord.h:1325:33: note: in implicit copy constructor for 'absl::FunctionRef<void (absl::string_view)>' first required here
    return ForEachChunkAux(rep, callback);
                                ^
1 error generated.
```

Somehow this code is built with `[-Werror,-Wdeprecated-copy]`, causing the warning to fail as an error.

Note that CLang was also changed from version 13.0 to 14.0.

--- a/bazel/repository.bzl
+++ b/bazel/repository.bzl
@@ -54,10 +54,10 @@ def opentelemetry_cpp_deps():
     maybe(
         http_archive,
         name = "com_google_absl",
-        sha256 = "dd7db6815204c2a62a2160e32c55e97113b0a0178b2f090d6bab5ce36111db4b",
-        strip_prefix = "abseil-cpp-20210324.0",
+        sha256 = "4208129b49006089ba1d6710845a45e31c59b0ab6bff9e5788a87f55c5abd602",
+        strip_prefix = "abseil-cpp-20220623.1",
         urls = [
-            "https://github.com/abseil/abseil-cpp/archive/20210324.0.tar.gz",
+            "https://github.com/abseil/abseil-cpp/archive/refs/tags/20220623.1.tar.gz",
         ],
     )
 

--- a/bazel/repository.bzl
+++ b/bazel/repository.bzl
@@ -54,7 +54,7 @@ def opentelemetry_cpp_deps():
     maybe(
         http_archive,
         name = "com_google_absl",
-        sha256 = "4208129b49006089ba1d6710845a45e31c59b0ab6bff9e5788a87f55c5abd602",
+        sha256 = "91ac87d30cc6d79f9ab974c51874a704de9c2647c40f6932597329a282217ba8",
         strip_prefix = "abseil-cpp-20220623.1",
         urls = [
             "https://github.com/abseil/abseil-cpp/archive/refs/tags/20220623.1.tar.gz",

